File: 16.3SumClosest.cpp
class Solution
{
public:
    int threeSumClosest(vector<int> &nums, int target)
    {
        int n = nums.size();
        int min_difference = INT_MAX;
        int result = INT_MAX;
        int left, right, difference, res;
        sort(nums.begin(), nums.end());

        for (int i = 0; i < n; i++)
        {
            left = i + 1;
            right = n - 1;

            while (left < right)
            {
                res = nums[i] + nums[left] + nums[right];
                difference = abs(res - target);

                if (difference < min_difference)
                {
                    min_difference = difference;
                    result = res;
                }

                if (res > target)
                {
                    right--;
                }
                else if (res < target)
                {
                    left++;
                }
                else
                {
                    return res;
                }
            }
        }

        return result;
    }
};

File: .py/16.3SumClosest.py
class Solution:
    def threeSumClosest(self, nums: List[int], target: int) -> int:
        nums = sorted(nums)
        result = float('inf')
        min_difference = float('inf')
        n = len(nums)
        
        for i in range(n):
            left, right = i + 1, n - 1
            while left < right:
                res = nums[i] + nums[left] + nums[right]
                difference = abs(res - target)
                
                if difference < min_difference:
                    min_difference = difference
                    result = res
                
                if res > target:
                    right -= 1
                elif res < target:
                    left += 1
                else:
                    return res  # res == target
        
        return result